src/tvdb/client.py

```diff
@@ -12,94 +12,149 @@
+        if isinstance(self.data, SearchResult) and self.data.image_url:
+            self.image_url = URL(self.data.image_url)
+        # that not isn't needed but is there for clarity and for pyright
+        elif not isinstance(self.data, SearchResult) and self.data.image:
+            self.image_url = URL(self.data.image)
+
+        if isinstance(self.data, SearchResult):
+            if self.data.translations and self.data.translations.root:
+                self.name_eng = self.data.translations.root.get("eng")
+            if self.data.overviews and self.data.overviews.root:
+                self.overview_eng = self.data.overviews.root.get("eng")
+        else:
+            if self.data.aliases:
+                self.name_eng = next(alias for alias in self.data.aliases if alias.language == "eng").name
+            if isinstance(self.data, (SeriesExtendedRecord, MovieExtendedRecord)) and self.data.translations:
+                if self.data.translations.name_translations:
+                    self.name_eng = next(
+                        translation.name
+                        for translation in self.data.translations.name_translations
+                        if translation.language == "eng"
+                    )
+                if self.data.translations.overview_translations:
+                    self.overview_eng = next(
+                        translation.overview
+                        for translation in self.data.translations.overview_translations
+                        if translation.language == "eng"
+                    )
+
+    @property
+    def bilingual_name(self) -> str | None:
+        if self.name == self.name_eng:
+            return self.name
+        return f"{self.name} ({self.name_eng})"
 
     @property
     def id(self) -> int | str | None:
         return self._id
 
     @id.setter
-    def id(self, value: int | str) -> None:
-        self._id = int(str(value).split("-")[1])
-
-    def __call__(self, media_id: str) -> "_Media":
-        self.id = media_id
-        return self
-
-    @abstractmethod
-    async def search(self, search_query: str, limit: int = 1) -> list[Any]: ...
+    def id(self, value: int | str | None) -> None:
+        if value:
+            self._id = int(str(value).split("-")[1])
+        else:
+            self._id = None
 
+    @classmethod
     @abstractmethod
-    async def fetch(self, *, extended: bool = False) -> Any: ...
-
-
-class Series(_Media):
-    """Class to interact with the TVDB API for series."""
+    async def fetch(cls, media_id: int | str, *, client: "TvdbClient", extended: bool = False) -> "_Media": ...
 
-    @override
-    async def search(self, search_query: str, limit: int = 1) -> list[SearchResult]:
-        """Search for a series in the TVDB database.
 
-        :param search_query:
-        :param limit:
-        :return:
-        """
-        return await self.client.search(search_query, "series", limit)
+class Movie(_Media):
```

Comment:
Decorate with `typing.final` to enforce the abstract methods being overridden (same for `Series` class)

Reply:
You mean
````py
  @final
    @abstractmethod
    @classmethod
    async def fetch(cls, media_id: int | str, *, client: "TvdbClient", extended: bool = False) -> "_Media": ...

```

Reply:
No, I mean: ```python
@final
class Movie(_Media):
    ...
 ```